code/__DEFINES/milla.dm

```diff
@@ -0,0 +1,129 @@
+// milla.dm - DM API for milla extension library
+
+// Default automatic MILLA detection.
+// Look for it in the build location first, then in `.`, then in standard places.
+
+/* This comment bypasses grep checks */ /var/__milla
+
+/proc/__detect_milla()
+	if(world.system_type == UNIX)
+		// First check if it's built in the usual place.
+		if(fexists("./milla/target/i686-unknown-linux-gnu/release/libmilla.so"))
```

Comment:
Also, how are we planning on building this? Are we going to expect contributors to cargo build it themselves, or were you thinking of maybe having ci build it for us?

Reply:
For users, it's the same as rust_g. Linux users build it, windows users get a checked-in DLL.

For people changing MILLA in a dev environment, they'll obviously build their own copy, and this code should prefer that over the pre-built DLL on Windows.

For PRs, my personal preference is that we get a bot set up that will re-build the DLL from source, and run that similar to !merge_upstream when the PR is ready to merge or get TM'd. The submitter wouldn't need to handle the DLL themselves that way, and if they did try anything fishy, the bot would clobber it.

Reply:
It looks like CI might currently be failing as well due to this not being checked in, it seems like something we might need to work out for this PR

Reply:
We should have handling for rust_g already, since that also doesn't leave a `.so` in the main directory. Just need to copy that and adjust for MILLA.

Reply:
> We should have handling for rust_g already, since that also doesn't leave a `.so` in the main directory. Just need to copy that and adjust for MILLA.

This is handled by a github release. For this one instance ill allow a `tools/ci/libmilla_ci.so`, just to avoid having to release-juggle it.

Reply:
> For this one instance ill allow a `tools/ci/libmilla_ci.so`, just to avoid having to release-juggle it.

Added and used `#ifdef CIBUILDING`.